### DISCUSSION TITLE
How to return predicted image and show and save in FastAPI

### DISCUSSION THREAD

**Discussion by pure-rgb** [fastapi/fastapi#6489] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

I mainly follow a web tutorial. I have a segmentation model that predicts a mask image. In FaskAPI, I'm a bit confused about how to save and show the model's predicted outcome. 

Here is the minimum code. 

 ```python
app = FastAPI()
model = # full model with weights 

@app.get('/')
def read_root():
    return {'hello':'world'}

@app.post('/prediction/')
async def prediction_route(file:UploadFile = File(...)):
    contents = await file.read()
    pil_image = Image.open(io.BytesIO(contents))
    pil_image = pil_image.resize((224, 224))

    if pil_image.model == 'RGBA':
        pil_image = pil_image.convert('RGB')
    
    numpy_image = np.array(pil_image) / 255.0
    numpy_image = np.expand_dims(numpy_image, axis=0)

    preds_image = model.predict(numpy_image) 
    preds_image = (preds_image.argmax(axis=-1)

    return FileResponse(f"{preds_image}.png")
```

### Description

It responded 

```
Response body
Download
{
  "detail": "model"
}
```

### Operating System

Windows

### FastAPI Version

0.70.1

### Python Version

3.9

### Additional Context

In the following 

![image](https://user-images.githubusercontent.com/45315076/148436649-6672913f-a1f2-4109-b358-0a62f4a1913c.png)

I tried but couldn't make it work. I was hoping to save the input image and show it somehow if possible.

**Comment by pure-rgb** [comment]
I found this https://stackoverflow.com/questions/55873174/how-do-i-return-an-image-in-fastapi but too many ways that confused me. I tried one/two solutions but didn't make it work.

**Comment by harunyasar** [comment, marked as answer]
AFAIK, the `FileReponse` class expects a file path in the filesystem, so you may need to store your predicted image somewhere.  But if you don't want to store this result, then a better option would be to use `StreamingResponse` by passing the content generator as the first argument of this streaming response class. You can find a better example and explanation in FastAPI documentation.
